diff -u -r -bB --no-dereference plumed-2.7.2/src/core/Atoms.cpp plumed-2.7.2_mod/src/core/Atoms.cpp
--- src/core/Atoms.cpp	2021-07-27 06:24:03.000000000 -0700
+++ src/core/Atoms.cpp	2021-10-06 15:56:35.000000000 -0700
@@ -74,6 +74,32 @@
   }
 }
 
+void Atoms::getShuttleAtomCount(int*p) {
+  std::set<AtomNumber> shuttle;
+  for(unsigned i=0; i<actions.size(); i++) {
+    if(!actions[i]->getUnique().empty()) {
+      shuttle.insert(actions[i]->getUnique().begin(), actions[i]->getUnique().end());
+    }
+  }
+  std::cout << "PLUMED: Shuttle size: ";
+  std::cout << shuttle.size();
+  std::cout << '\n';
+  *p = shuttle.size();
+}
+
+void Atoms::getShuttleAtoms(int*p) {
+  std::set<AtomNumber> shuttle;
+  for(unsigned i=0; i<actions.size(); i++) {
+    if(!actions[i]->getUnique().empty()) {
+      shuttle.insert(actions[i]->getUnique().begin(), actions[i]->getUnique().end());
+    }
+  }
+  std::vector<int> shuttle_vector;
+  shuttle_vector.resize(shuttle.size());
+  std::transform(shuttle.begin(), shuttle.end(), shuttle_vector.begin(), [](const AtomNumber &n){return n.serial();});
+  std::copy(shuttle_vector.begin(), shuttle_vector.end(), p);
+}
+
 void Atoms::startStep() {
   collectEnergy=false; energyHasBeenSet=false; positionsHaveBeenSet=0;
   massesHaveBeenSet=false; chargesHaveBeenSet=false; boxHasBeenSet=false;
diff -u -r -bB --no-dereference plumed-2.7.2/src/core/Atoms.h plumed-2.7.2_mod/src/core/Atoms.h
--- src/core/Atoms.h	2021-07-27 06:24:03.000000000 -0700
+++ src/core/Atoms.h	2021-10-06 15:59:19.000000000 -0700
@@ -182,6 +182,9 @@
   void setAtomsContiguous(int);
   void setAtomsNlocal(int);
 
+  void getShuttleAtomCount(int*);
+  void getShuttleAtoms(int*);
+
   void startStep();
   void setEnergy(void*);
   void setBox(void*);
diff -u -r -bB --no-dereference plumed-2.7.2/src/core/PlumedMain.cpp plumed-2.7.2_mod/src/core/PlumedMain.cpp
--- src/core/PlumedMain.cpp	2021-07-27 06:24:03.000000000 -0700
+++ src/core/PlumedMain.cpp	2021-10-06 15:58:11.000000000 -0700
@@ -184,6 +184,14 @@
       const auto it=word_map.find(words[0]);
       if(it!=word_map.end()) iword=it->second;
       switch(iword) {
+      case cmd_getShuttleAtomCount:
+        CHECK_INIT(initialized,word);
+        atoms.getShuttleAtomCount(static_cast<int*>(val));
+        break;
+      case cmd_getShuttleAtoms:
+        CHECK_INIT(initialized,word);
+        atoms.getShuttleAtoms(static_cast<int*>(val));
+        break;
       case cmd_setBox:
         CHECK_INIT(initialized,word);
         CHECK_NOTNULL(val,word);
diff -u -r -bB --no-dereference plumed-2.7.2/src/core/PlumedMainEnum.inc plumed-2.7.2_mod/src/core/PlumedMainEnum.inc
--- src/core/PlumedMainEnum.inc	2021-10-04 20:10:17.000000000 -0700
+++ src/core/PlumedMainEnum.inc	2021-10-06 15:59:47.000000000 -0700
@@ -1,3 +1,5 @@
+cmd_getShuttleAtomCount,
+cmd_getShuttleAtoms,
 cmd_setBox,
 cmd_setPositions,
 cmd_setMasses,
diff -u -r -bB --no-dereference plumed-2.7.2/src/core/PlumedMainMap.inc plumed-2.7.2_mod/src/core/PlumedMainMap.inc
--- src/core/PlumedMainMap.inc	2021-10-04 20:10:17.000000000 -0700
+++ src/core/PlumedMainMap.inc	2021-10-06 15:59:47.000000000 -0700
@@ -1,3 +1,5 @@
+  {"getShuttleAtomCount",cmd_getShuttleAtomCount},
+  {"getShuttleAtoms",cmd_getShuttleAtoms},
   {"setBox",cmd_setBox},
   {"setPositions",cmd_setPositions},
   {"setMasses",cmd_setMasses},
